ElectronClient/gui/NoteText.jsx

```diff
@@ -995,6 +995,23 @@ class NoteTextComponent extends React.Component {
 			const that = this; // The "this" within the function below refers to something else
 			this.editor_.editor.getSession().getMode().getNextLineIndent = function(state, line) {
 				const ls = that.state.lastKeys;
+
+				// Detects empty List Element
+				// Solves: https://github.com/laurent22/joplin/issues/2614
+
+				let regexer = /^[\d]+[.] $/;
+				if (line == '- ' || line == '* ' || line == '- [ ] ' || line== '- [x] '|| regexer.test(line)) {
+					let s = that.currentTextOffset();
+
+					const s1 = that.state.note.body.substr(0, s-(line.length+1));
+					const s2 = that.state.note.body.substr(s);
+					console.log(s1);
+					console.log(s2);
+					shared.noteComponent_change(that, 'body', s1 + s2);
```

Comment:
The containing function is called `getNextLineIndent` so it should not have any side-effect, and in fact the note doesn't change so the "change" event should not be triggered. I'm not sure how to best implement what you want to do, but this feels a bit like a hack.

Reply:
Its kind of hack... But it works!

Reply:
Under the same function ( _getNextLineIndent_ ), we are checking if the next line should start with a List Element character (line 1018-1023). So I thought this was the best place to deal with the issue.
![paoa](https://user-images.githubusercontent.com/41316051/75648064-03a07d80-5c75-11ea-8e0f-07d6b961e87f.png)

Reply:
I understand you are worried about the side effect.
I will try something else!
@laurent22